src/ngraph/pass/nop_elimination.cpp

```diff
@@ -69,25 +69,54 @@ static bool eliminate_sum(const std::shared_ptr<Node>& node)
     }
     return false;
 }
-
+/*************************
+ * Eliminates 3 types of spines with converts:
+ *
+ * 1. [ op1--> convert2(type A->B)--> ReduceMin--> convert1(type B->A)--> op2 ] to
+ *                    [ op1--> ReduceMin--> op2 ]
+ * 2. [ op3--> convert1-->  NonZero--> op4 ] to
+ *                    [ op3--> NonZero--> op4 ]
+ * 3. [ op5--> convert1--> convert2--> op6 ] to
+ *                    [op5--> convert2--> op6 ]
+ *
+ **************************/
 static bool eliminate_convert(const std::shared_ptr<Node>& node)
 {
+    auto convert = as_type_ptr<opset3::Convert>(node);
+    auto destination_type = convert->get_destination_type();
+    auto input_op = convert->input_value(0);
+
+    // case 1
+    if (is_type<opset3::ReduceMin>(input_op.get_node()))
+    {
+        auto convert_2 = input_op.get_node()->input_value(0).get_node();
+        if (is_type<opset3::Convert>(convert_2) && convert_2->get_users().size() == 1 &&
+            destination_type == convert_2->get_input_element_type(0))
+        {
```

Comment:
Not valid for all type combinations. (A, B) as (float, double), (bf16|fp16, float|double), (intx, int y>=x), (uintx, uinty>=x), (uintx, inty>x), and (A,A) are valid.

Reply:
Should we just limit to the following types, which is just the MaskRCNN use cases:
For case 1: ReduceMin -> i64 ->i32->i64

Reply:
updated.